File: leetcode3/AccountBalanceAfterRoundedPurchase/a.cpp
/* Copyright (C) 2023 by iamslash */

#include <cstdio>
#include <cmath>

using namespace std;

// 2ms 42.86% 6MB 57.14%
class Solution {
public:
  int accountBalanceAfterPurchase(int purchaseAmount) {
    double amt = purchaseAmount * 0.1;
    return 100 - round(amt) * 10;
  }
};

// 3ms 42.86% 6MB 57.14%
class Solution {
public:
  int accountBalanceAfterPurchase(int purchaseAmount) {
    if (purchaseAmount % 10) {
      int digit = purchaseAmount % 10;
      if (purchaseAmount % 10 < 5) {        
        purchaseAmount -= digit;
      } else {
        purchaseAmount += 10 - digit;
      }
    }
    return 100 - purchaseAmount;
  }
};

int main() {
  return 0;
}

File: leetcode3/AccountBalanceAfterRoundedPurchase/a.swift

// 0ms 100.00% 13.9MB 100.00%
// math
// O(1) O(1)
class Solution {
    func accountBalanceAfterPurchase(_ purchaseAmount: Int) -> Int {
        var amt = purchaseAmount;
        let dgt = amt % 10;
        if (dgt > 0) {
            amt -= dgt;
            if (dgt >= 5) {
                amt += 10;
            }
        }
        return 100 - amt;
    }
}
